node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -0,0 +1,246 @@
+    seedOutput = BS.replicate seedSize (1 :: Word8)
+
+    (sk, addr, txOutAuxInput) = generateTxOutAux 1000000 seedInput
+    (_, txOutAuxOutput)       = generateRedeemTxOutAux 1 seedOutput
+
+    utxo = one (TxInUtxo (unsafeIntegerToTxId 0) 0, txOutAuxInput)
+    signers = one (fakeSigner sk, addr)
+    outputs = one txOutAuxOutput
+    addrData = ()
+
+----------------------------------------------------------------------------
+-- Helpers
+----------------------------------------------------------------------------
+
+data CreateMTxParams = CreateMTxParams
+    { cmpUtxo     :: !Utxo
+    , cmpSigners  :: !(NonEmpty (SafeSigner, Address))
+    , cmpOutputs  :: !TxOutputs
+    , cmpAddrData :: !(AddrData TxpTestMode)
+    }
+
+makeManyUtxoTo1Params :: Integer -> Integer -> Integer -> CreateMTxParams
+makeManyUtxoTo1Params numFrom amountEachFrom amountTo = CreateMTxParams {..}
+  where
+    seedInput  = BS.replicate seedSize (0 :: Word8)
+    seedOutput = BS.replicate seedSize (1 :: Word8)
+
+    (sk, addr, txOutAuxInput) = generateTxOutAux amountEachFrom seedInput
+    (_, _, txOutAuxOutput)    = generateTxOutAux amountTo       seedOutput
+
+    cmpUtxo = M.fromList
+        [(TxInUtxo (unsafeIntegerToTxId 0) (fromIntegral k), txOutAuxInput) |
+            k <- [0..numFrom-1]]
+    cmpSigners = one (fakeSigner sk, addr)
+    cmpOutputs = one txOutAuxOutput
+    cmpAddrData = ()
+
+makeManyAddressesToManyParams :: Word8 -> Integer -> Word8 -> Integer -> CreateMTxParams
+makeManyAddressesToManyParams numFrom amountEachFrom numTo amountEachTo = CreateMTxParams {..}
+  where
+    seedsInput = [BS.replicate seedSize v | v <- [0..numFrom-1]]
+    (cmpSignersList, txOutAuxInputs) = unzip
+        [((fakeSigner sk, addr), txOutAux) |
+            (sk, addr, txOutAux) <- map (generateTxOutAux amountEachFrom) seedsInput]
+
+    cmpSigners = NE.fromList cmpSignersList
+
+    seedsOutput = [BS.replicate seedSize v | v <- [numFrom..numFrom+numTo-1]]
+    txOutAuxOutputs = [txOutAux |
+        (_, _, txOutAux) <- map (generateTxOutAux amountEachTo) seedsOutput]
+
+    cmpUtxo = M.fromList [(TxInUtxo (unsafeIntegerToTxId $ fromIntegral k) 0, txOutAux) |
+        (k, txOutAux) <- zip [0..numFrom-1] txOutAuxInputs]
+    cmpOutputs = NE.fromList txOutAuxOutputs
+    cmpAddrData = ()
+
+ensureTxMakesSense
+  :: HasCoreConstants
+  => TxWithSpendings -> Utxo -> TxOutputs -> TxpTestProperty ()
+ensureTxMakesSense (_, neTxOut) utxo _ = do
+    unless (S.fromList txOutUsed `S.isSubsetOf` S.fromList txOutAvailable) $
+        stopProperty $
+            sformat ("Used some inputs that were not available!\n"%
+                    "Available: "%shown%"\n"%
+                    "Used: "%shown
+                    ) txOutAvailable txOutUsed
+  where
+    txOutAvailable = map toaOut $ M.elems utxo
+    txOutUsed = NE.toList neTxOut
+
+unsafeIntegerToTxId :: Integer -> TxId
+unsafeIntegerToTxId n =
+    unsafeFromJust $ rightToMaybe $ decodeHash $
```

Comment:
I would recommend using `leftToPanic "something"` instead of `unsafeFromJust . rightToMaybe` to have more descriptive error message when impossible happens.

Reply:
Didn't know about that helper. Fixed.